.github/workflows/cypress.yaml

```diff
@@ -25,12 +25,13 @@ jobs:
           path: care
 
       - name: Run docker compose up on care 🐳
-        run: cd care && touch .env && make docker_config_file=docker-compose.pre-built.yaml up && cd .. && sleep 60s
-        # Voluntarily kept 60 seconds delay to wait for migrations to complete.
-
-      - name: Run Django collectstatic and load dummy data on care 🐍
         run: |
-          docker exec care python manage.py load_dummy_data
+          cd care  
+          make docker_config_file=docker-compose.pre-built.yaml up 
+          sleep 60s
+          docker compose exec backend bash -c "python manage.py load_dummy_data"
+          cd ..
+        # Voluntarily kept 60 seconds delay to wait for migrations to complete.
```

Comment:
```suggestion
          cd care  
          make docker_config_file=docker-compose.pre-built.yaml up 
          until docker-compose -f docker-compose.pre-built.yaml exec -T backend bash -c "python manage.py showmigrations | grep -q '\[ \]'"; do
            >&2 echo "Migrations are not yet applied - sleeping"
            sleep 5
          done
          docker compose exec backend bash -c "python manage.py load_dummy_data"
          cd ..
```

----
Instead of this `hack`, how about we wait for the migrations to finish, detect that then proceed?

We could poll the output of `python manage.py showmigrations | grep '\[ \]'` every 5 seconds and if empty, then know for certain that migrations are completed. 

Sample output while migrations are running in the container

```
[root@2b4804e753a8] [~/care] ± testing {4} ?:1                                                                                                   [15:10:05]
> python manage.py showmigrations | grep '\[ \]'
 [ ] 0009_healthfacility
 [ ] 0368_populate_medibase
 [ ] 0366_auto_20230627_1806
 [ ] 0369_merge_0366_auto_20230627_1806_0368_populate_medibase
 [ ] 0367_auto_20230704_1646
 [ ] 0370_merge_20230705_1500
 [ ] 0371_metaicd11diagnosis_chapter_and_more
 [ ] 0372_assetavailabilityrecord
 [ ] 0373_remove_patientconsultation_hba1c
 [ ] 0374_historicalpatientregistration_abha_number_and_more (5 squashed migrations)
 [ ] 0375_alter_dailyround_resp
 [ ] 0376_auto_20230807_2148
 [ ] 0376_alter_ambulance_owner_phone_number_and_more
 [ ] 0377_merge_20230809_0009
 [ ] 0378_consultationbedasset_consultationbed_assets
 [ ] 0379_rename_prescribed_medication_patientconsultation_treatment_plan
 [ ] 0380_patientnotes_user_type
 [ ] 0381_merge_ip_op_no
 [ ] 0382_assetservice_remove_asset_last_serviced_on_and_more
 [ ] 0379_consultationbed_privacy
 [ ] 0383_merge_20230905_1213
 [ ] 0384_alter_notification_event


[root@2b4804e753a8] [~/care] ± testing {4} ?:1                                                                                                   [15:10:08]
> python manage.py showmigrations | grep '\[ \]'
 [ ] 0380_patientnotes_user_type
 [ ] 0381_merge_ip_op_no
 [ ] 0382_assetservice_remove_asset_last_serviced_on_and_more
 [ ] 0379_consultationbed_privacy
 [ ] 0383_merge_20230905_1213
 [ ] 0384_alter_notification_event


[root@2b4804e753a8] [~/care] ± testing {4} ?:1                                                                                                   [15:10:16]
130 > python manage.py showmigrations | grep '\[ \]'
```

[Note: suggested workflow not tested on Github actions yet]

Reply:
Well, missed by a minute. 😅 We can consider this in the future then, cc: @rithviknishad @khavinshankar

Reply:
That would be great

Reply:
@Ashesh3 add this in a new pr